app/javascript/autocompletion/complete_detail.ts

```diff
@@ -0,0 +1,99 @@
+import * as dicts from "./detail_dicts"
+
+/** 補完候補 */
+type CompletionCandidate = {
+  /** このキーワードが含まれていたら補完を行う */
+  keywords: Array<string>
+  /** 補完に使う文字列 */
+  completion: string
+}
+
+/** 補完辞書 */
+type CompletionDict = Array<CompletionCandidate>
+
+/**
+ * 指定した複数文字列のいづれか1つでも含まれるかをチェックする
+ *
+ * String.includesの複数候補版。
+ *
+ * @param target 検索対象の文字列
+ * @param keywords 検索する複数キーワード
+ * @returns 検索対象にいづれかのキーワードが1つでも含まれればtrue
+ */
+function includesAny(target: string, keywords: Array<string>): boolean {
+  return keywords
+    .map((word) => {
+      return target.includes(word)
+    })
+    .includes(true)
```

Comment:
```suggestion
    .some((word) => {
      return target.includes(word)
    })
```
１つでも true が見つかった時点でスキャンを終えられる some のほうが効率が良いでしょう
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
欲しかった関数だ！！
ありがとう！！

名前付けが言語によって違うので検索ミスりがち。